### ISSUE TITLE
`metricset.name: process` is not generated for metricbeat run through docker.

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/beats#41394]
**Kibana Build details:**
```
VERSION: 8.16.0-BC1
BUILD: 79314
COMMIT: 5575428dd3aef69366cddb4ccf07a2a26d30ce48
```
Artifact: docker.elastic.co/staging/metricbeat:8.16.0-e8d5928a 

**Host OS:** Rhel 9 and Ubuntu 24

**Preconditions:**
1. 8.16.0-BC1 Kibana cloud environment should be available.
2. Docker should be setup.

**Steps to reproduce:**
1. Run below command
```
sudo docker run --label co.elastic.metrics/module=system \
--mount type=bind,source=/proc,target=/hostfs/proc,readonly \
--mount type=bind,source=/sys/fs/cgroup,target=/hostfs/sys/fs/cgroup,readonly \
--mount type=bind,source=/,target=/hostfs,readonly \
--mount type=bind,source=/var/run/dbus/system_bus_socket,target=/hostfs/var/run/dbus/system_bus_socket,readonly \
--env DBUS_SYSTEM_BUS_ADDRESS='unix:path=/hostfs/var/run/dbus/system_bus_socket' \
--net=host --cgroupns=host docker.elastic.co/staging/metricbeat:8.16.0-e8d5928a metricbeat -e -E output.elasticsearch.hosts='https://6***************************d.com:443' \
-E output.elasticsearch.username='elastic' \
-E output.elasticsearch.password='******************' \
-d '*'
```
2. Observe metricbeat data is generated. However, `metricset.name: process` is not generated under Discover tab.
3. Now run below command:
```
sudo docker run --label co.elastic.metrics/module=system \
--mount type=bind,source=/proc,target=/hostfs/proc,readonly \
--mount type=bind,source=/sys/fs/cgroup,target=/hostfs/sys/fs/cgroup,readonly \
--mount type=bind,source=/,target=/hostfs,readonly \
--mount type=bind,source=/var/run/dbus/system_bus_socket,target=/hostfs/var/run/dbus/system_bus_socket,readonly \
--env DBUS_SYSTEM_BUS_ADDRESS='unix:path=/hostfs/var/run/dbus/system_bus_socket' \
--net=host docker.elastic.co/staging/metricbeat:8.16.0-e8d5928a metricbeat -e -E output.elasticsearch.hosts='https://6***************************d.com:443' \
-E output.elasticsearch.username='elastic' \
-E output.elasticsearch.password='******************' \
-d '*'
```
4. Still observe `metricset.name: process` is not generated under Discover tab.

**Logs:**
[docker-metricbeat.txt](https://github.com/user-attachments/files/17488604/docker-metricbeat.txt)

**Expected Result:**
`metricset.name: process` should be generated for metricbeat run through docker.

**Screen Recording:**

https://github.com/user-attachments/assets/27c5a1c8-e4e8-47ef-9288-ad9e9b87557d

**Feature:**
https://github.com/elastic/beats/issues/39900

**Comment by amolnater-qasource** [comment]
@muskangulati-qasource Please review.

**Comment by muskangulati-qasource** [comment]
Secondary review is Done for this ticket!

**Comment by pierrehilbert** [comment]
Hey @amolnater-qasource could you please confirm this is working with 8.15.3?

**Comment by amolnater-qasource** [comment]
Hi @pierrehilbert 

Thank you for looking into this issue.
We have revalidated this with `docker.elastic.co/beats/metricbeat:8.15.3` and found it **reproducible** there too.
- `metricset.name: process` is not generated for metricbeat run through docker. 

Further for confirmation we have tested the same with 8.15.2, 8.15.1 and 8.15.0 and had below observations:
- 8.15.2: Reproducible.
- 8.15.1: Not Reproducible
- 8.15.0: Not Reproducible

**Screen Recording:**


https://github.com/user-attachments/assets/9a24e7c7-e058-4e6f-9f3d-64cfda0702b2


Please let us know if anything else is required from our end.
Thanks!!

**Comment by pierrehilbert** [comment]
I'm looking at changes from 8.15.1 to 8.15.2 but didn't find anything for now.
Maybe @cmacknz would have an idea

**Comment by pierrehilbert** [comment]
@amolnater-qasource what version of the system integration are you using?

**Comment by cmacknz** [comment]
Agree, none of the commits in https://github.com/elastic/beats/compare/v8.15.1...v8.15.2 look like they should be related.

**Comment by amolnater-qasource** [comment]
Hi @pierrehilbert 

8.16.0 metricbeat was tested on the same 8.16.0 BC1 kibana cloud environment.

For 8.15.x versions we used the same 8.15.3 kibana cloud environment. 
System integration v1.61.1 is available on 8.15.3.

**Build details:**
```
VERSION: 8.15.3
BUILD: 76711
COMMIT: 3933429968aafb1ba31319fc38649d0f974044bf
```

Please let us know if anything else is required from our end.
Thanks!!

**Comment by amolnater-qasource** [comment]
Hi Team,

We have revalidated this issue on 8.17.0 BC2 kibana cloud environment and found it **fixed now**.

**Observations:**
- `metricset.name: process` is generated for metricbeat run through docker.

**Host OS:**
Ubuntu 24.04 and Rhel 9

**Screenshots:**
**Ubuntu 24.04:**
![Image](https://github.com/user-attachments/assets/3ecab198-0ddd-43e7-93e3-1b6570e60bc2)

**Rhel9:**
![Image](https://github.com/user-attachments/assets/e895f490-d76a-4cda-ab02-0229fdc133f8)


Hence, we are closing this issue and marking as **QA:Validated**.

Thanks!
